### PR TITLE
Proto revision

### PR DESCRIPTION
**Pull request by wtong98**
Update the collector backend code with changes made to the protobufs. This PR also updates the model system to include error checking when converting to protos.

### CODE REVIEW

File: extension/dynamicconfig/dynamicconfig.go [hunk 1]
```diff
@@ -43,11 +43,11 @@ func newServer(config Config, logger *zap.Logger) (*dynamicConfigExtension, erro
 	return de, nil
 }
 
+// TODO: debug ReportFatalError hanging
```
Comment (vmingchen) [inline extension/dynamicconfig/dynamicconfig.go:46]: Please remove.

File: extension/dynamicconfig/model/collection_period.go [hunk 2]
```diff
@@ -19,20 +19,65 @@
...
 	"hash/fnv"
+	"strconv"
 )
 
 type CollectionPeriod string
 
-func (period CollectionPeriod) Proto() pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod {
-	interval := pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod_value[string(period)]
-	return pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod(interval)
+func (period CollectionPeriod) Proto() (int32, error) {
```
Comment (vmingchen) [inline extension/dynamicconfig/model/collection_period.go:30]: Let's add a TODO here to see if there is a library for parsing time duration.

1s = 1s
1m  = 60s
1h = 3600s

File: extension/dynamicconfig/testdata/schedules_bad.yaml [hunk 3]
```diff
@@ -1,19 +1,12 @@
...
-              - InclusionPatterns:
-                    - Equals: "woot/woot/woot"
-                Period: "DAY_7"
-
     - Resource:
-          - "namespace: weather-forecast"
-          - "deployment-id: 5564279"
+      Schedules:
+          - InclusionPattern:
+                - StartsWith: "/missing/s"
```
Comment (vmingchen) [inline extension/dynamicconfig/testdata/schedules_bad.yaml:5]: Add a comment

File: extension/dynamicconfig/testdata/schedules_improper_pattern.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+ConfigBlocks:
+    - Resource:
+      Schedules:
+          - InclusionPatterns:
+                - StartsWith: "*"
```
Comment (vmingchen) [inline extension/dynamicconfig/testdata/schedules_improper_pattern.yaml:5]: Add a comment

File: extension/dynamicconfig/dynamicconfig.go (hunk 1 above)
Comment (wtong98) [reply extension/dynamicconfig/dynamicconfig.go:46]: Done

File: extension/dynamicconfig/model/collection_period.go (hunk 2 above)
Comment (wtong98) [reply extension/dynamicconfig/model/collection_period.go:30]: Done

File: extension/dynamicconfig/testdata/schedules_bad.yaml (hunk 3 above)
Comment (wtong98) [reply extension/dynamicconfig/testdata/schedules_bad.yaml:5]: Done

File: extension/dynamicconfig/testdata/schedules_improper_pattern.yaml (hunk 4 above)
Comment (wtong98) [reply extension/dynamicconfig/testdata/schedules_improper_pattern.yaml:5]: Done

**Pull request merged**